packages/commonwealth/server/controllers/server_profiles_methods/get_community_members.ts

```diff
@@ -0,0 +1,304 @@
+    chains: string[];
+    addresses: string[];
+    stake_balances: string[];
+    last_active: string;
+  }>(`${sqlWithoutPagination}`, {
+    bind,
+    type: QueryTypes.SELECT,
+  });
+  const totalResults = fullResults.length;
+
+  if (includeStakeBalances) {
+    const stake = await this.models.CommunityStake.findOne({
+      where: { community_id: community.id },
+    });
+    if (!stake) {
+      throw new AppError(Errors.StakeNotFound);
+    }
+    const stakeholderGroup = await this.models.Group.findOne({
+      where: {
+        community_id: community.id,
+        is_system_managed: true,
+      },
+    });
+    if (!stakeholderGroup) {
+      throw new AppError(Errors.StakeholderGroup);
+    }
+    const node = await this.models.ChainNode.findByPk(community.chain_node_id);
+    if (!node) {
+      throw new AppError(Errors.ChainNodeNotFound);
+    }
+    const addresses = fullResults.map((p) => p.addresses).flat();
+    const balances = await contractHelpers.getNamespaceBalance(
+      community.namespace,
+      stake.stake_id,
+      node.eth_chain_id,
+      addresses,
+      node.url,
+    );
+    // add balances to profiles
+    for (const profile of fullResults) {
+      for (const address of profile.addresses) {
+        profile.stake_balances ||= [];
+        profile.stake_balances.push(balances[address] || '0');
+      }
+    }
+  }
+
+  const paginatedResults = fullResults
+    .slice()
+    .sort((a, b) => {
+      let comparison = 0;
+      switch (orderBy) {
+        case 'created_at':
+          comparison = moment(a.created_at).isAfter(b.created_at) ? 1 : -1;
+          break;
+        case 'profile_name':
+          comparison = a.profile_name.localeCompare(b.profile_name);
+          break;
+        default:
+          comparison = moment(a.last_active).isAfter(b.last_active) ? 1 : -1;
+      }
+      // Adjust sort order based on orderDirection
+      return orderDirection === 'DESC' ? -comparison : comparison;
+    })
+    .slice((page - 1) * limit, page * limit);
+
+  const profilesWithAddresses: Profile[] = paginatedResults.map(
+    (profile: any) => {
+      return {
+        id: profile.id,
+        user_id: profile.user_id,
+        profile_name: profile.profile_name,
+        avatar_url: profile.avatar_url,
+        addresses: profile.address_ids.map((_, i) => {
+          const address: any = {
+            id: profile.address_ids[i],
+            chain: profile.chains[i],
+            address: profile.addresses[i],
+          };
+          if (profile.stake_balances) {
+            address.stake_balance = profile.stake_balances?.[i] || '0';
+          }
+          return address;
+        }),
+        roles: [],
+        group_ids: [],
+      };
+    },
+  );
+
+  if (includeRoles) {
+    const profileAddressIds = profilesWithAddresses.reduce((acc, p) => {
+      const ids = p.addresses.map((addr) => addr.id);
+      return [...acc, ...ids];
+    }, []);
+
+    const roles = await findAllRoles(
```

Comment:
Since it makes front end integration easier, I guess it is not a huge deal, but this findAllRoles method is deprecated and we should not be using it. We can get the relavant information from the "Addresses" table. Also maybe instead of looping through a bunch of these objects to create them, ideally the query could look something like:

```
SELECT ... (Original query here) ...
json_agg(json_build_object(
'id': a.id::number,
'chain': a.community_id:text,
'address': a.address::text
)) as addresses
...
(Remainder of original query)
...
```

So this way we can just return this result as the correct object, and don't need to perform any synchronous operations to build it on the javascript end.